unload_databricks_data_to_s3.py

```diff
@@ -213,6 +214,55 @@ def get_partition_count(event_count: int, max_event_count_per_output_file: int)
     return max(1, math.ceil(event_count / max_event_count_per_output_file))
 
 
+def replace_double_slashes_with_single_slash(string: str) -> str:
+    while '//' in string:
+        string = string.replace('//', '/')
+    return string
+
+
+def move_spark_metadata_to_separate_s3_folder(s3_client: boto3.client, s3_uri_with_spark_metadata: str):
+    """
+    Lists all files in the s3_uri_with_spark_metadata directory and moves all spark metadata files (files without
+    '/part-' in the name) to a separate subdirectory under s3_uri_with_spark_metadata.
+
+    NOTE: Expects spark metadata files to be in the same directory as the data files and available at the time of the
+    function call
+
+    :param s3_client: boto3 client for s3
+    :param s3_uri_with_spark_metadata: s3 URI with spark metadata files
+    """
+    print(f'Moving spark metadata files to a separate subdirectory in {s3_uri_with_spark_metadata}')
+
+    if '://' not in s3_uri_with_spark_metadata:
+        raise ValueError(f'Invalid s3 URI: {s3_uri_with_spark_metadata}. Expected to contain "://".')
+    bucket, prefix = s3_uri_with_spark_metadata.split('://')[1].split('/', 1)
+    bucket = replace_double_slashes_with_single_slash(bucket)
+    prefix = replace_double_slashes_with_single_slash(prefix)
+    print(f'Identified bucket: {bucket}, prefix: {prefix}')
+
+    # List all files in the s3_path directory
+    response = s3_client.list_objects_v2(Bucket=bucket, Prefix=prefix, Delimiter='/')
```

Comment:
Will this return folders as well? If only files are returned, then we are good.

Reply:
only files. Same for listing objects in our Java repo using Amplitude's S3 wrapper